locust/runners.py

```diff
@@ -166,7 +171,17 @@ def kill_locusts(self, kill_count):
                     to_kill.append(user)
                     bucket.remove(l)
                     break
+
+        to_kill_ids = sorted([user.id for user in to_kill])
+        remaining_count = len(self.locusts) - kill_count
+        for g in self.locusts:
+            if g.args[0].id >= remaining_count:
+                g.args[0].id = to_kill_ids.pop()
+
         self.kill_locust_instances(to_kill)
+        if not sorted([g.args[0].id for g in self.locusts]) == list(range(len(self.locusts))):
+            logger.warning("Locust IDs are not consecutive.")
```

Comment:
When could this happen?

Reply:
Same comment as above. Just making sure everything is good after some Locusts were killed.

Should I remove these checks?